pkg/kubelet/prober/prober.go

```diff
@@ -298,17 +300,18 @@ func (eic execInContainer) Stop() {
 }
 
 func (eic execInContainer) Start() error {
-	return fmt.Errorf("unimplemented")
+	return nil
 }
 
 func (eic execInContainer) Wait() error {
-	return fmt.Errorf("unimplemented")
+	return nil
 }
 
 func (eic execInContainer) StdoutPipe() (io.ReadCloser, error) {
-	return nil, fmt.Errorf("unimplemented")
+	b, err := eic.run()
+	return ioutil.NopCloser(bytes.NewReader(b)), err
```

Comment:
Can you share more about the `NopCloser`? What makes it preferable here?

Reply:
Is that we are reading from `Stdout/Stderr`? So it doesn't actually want to close the file?

Reply:
yep, we are readying from Stdout/Stderr